testdata/BUILD

```diff
@@ -533,6 +534,14 @@ container_image(
     tars = [":passwd_tar"],
 )
 
+container_image(
+    name = "with_passwd_homedirs",
+    tars = [
+        ":passwd_tar",
```

Comment:
It looks like `passwd_file` rule will automatically produce an additional `%{name}-homedirs.tar` output file right? 

Looking at this example, we have `:passwd_tar` which is a label (a `pkg_tar` target), and we have `:passwd-homedirs.tar` which is a file (refer to the `passwd_file` output). This looks a little unnatural to me.  

Is it possible to refactor the `passwd_file` rule itself, or the `pkg_tar` rule in this test, to produce only on single target, so that `container_image` rule can refer to this single target without thinking about the output file? This also serves as a good example for users on how to use `passwd_file` rule.

Reply:
If there is a way to mark some rules as deprecate, we can  definitely deprecate `passwd_file` and spit out warnings to use `passwd_tar` instead.